phoenix-core/src/main/java/org/apache/phoenix/index/IndexMaintainer.java

```diff
@@ -883,6 +903,14 @@ public void readFields(DataInput input) throws IOException {
         dataEmptyKeyValueCF = Bytes.readByteArray(input);
         emptyKeyValueCFPtr = new ImmutableBytesPtr(Bytes.readByteArray(input));
         
+        int numIndexedExpressions = WritableUtils.readVInt(input);
```

Comment:
Yes, same behavior for an old client talking to new server. Will this be difficult?